deltacat/compute/compactor/compaction_session.py

```diff
@@ -163,7 +168,9 @@ def _execute_compaction_round(
     rebase_source_partition_high_watermark: Optional[int],
     enable_profiler: Optional[bool],
     metrics_config: Optional[MetricsConfig],
-    list_deltas_kwargs=Optional[Dict[str, Any]],
+    list_deltas_kwargs: Optional[Dict[str, Any]],
+    read_kwargs_provider: Optional[ReadKwargsProvider],
```

Comment:
Thanks for adding. Have you thought about allowing clients to pass in various ReadKwargsProviders for specific steps (hash bucketing, materialize)?

Reply:
Yeah, thought about that. Assuming, we have to read the pk the same way (enforcing schema) we read during materialize, I kept it as one arg for now. If we find a need we can add if required in future.